sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net6.0.cs

```diff
@@ -191,17 +192,77 @@ public partial struct DynamicJsonOptions
         public string Name { get { throw null; } }
         public Azure.Core.Dynamic.DynamicJson Value { get { throw null; } }
     }
+    public abstract partial class ObjectDocument : System.IDisposable
+    {
+        protected ObjectDocument() { }
+        public abstract T As<T>(object element);
+        public abstract void Dispose();
+        public abstract Azure.Core.Dynamic.ObjectElement GetIndexElement(object element, int index);
+        public abstract bool HasValue(object element);
+        public abstract void Set(object element, object value);
+        public abstract Azure.Core.Dynamic.ObjectElement SetProperty(object element, string name, object value);
+        public abstract string? ToString(object element);
+        public abstract bool TryGetArrayLength(object element, out int length);
+        public abstract bool TryGetBoolean(object element, out bool value);
+        public abstract bool TryGetDouble(object element, out double value);
+        public abstract bool TryGetInt64(object element, out long value);
+        public abstract bool TryGetProperty(object element, string name, out Azure.Core.Dynamic.ObjectElement value);
+        public abstract bool TryGetPropertyNames(object element, out System.Collections.Generic.IEnumerable<string> enumerator);
+        public abstract bool TryGetString(object element, out string? value);
+        public abstract void WriteTo(object element, System.IO.Stream stream);
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ObjectElement
```

Comment:
Since this will be boxed every single time a method on it is called, does it still make sense to make it a struct?

Reply:
Yeah, I was hoping we'd find a way to do it without boxing - if we can't, then I think it does make sense for it to be a reference type instead.  In that case, I think it might make sense to make ObjectElement the abstract class rather than ObjectDocument, since that's the one that DynamicData calls into.  I'll have a look at that in the next iteration.